src/app/modules/form/public-form/__tests__/public-form.controller.spec.ts

```diff
@@ -389,4 +423,551 @@ describe('public-form.controller', () => {
       expect(mockRes.redirect).toHaveBeenCalledWith(expectedRedirectPath)
     })
   })
+
+  describe('handleGetPublicForm', () => {
+    const MOCK_FORM_ID = new ObjectId().toHexString()
+    const MOCK_USER_ID = new ObjectId().toHexString()
+    const MOCK_USER = {
+      _id: MOCK_USER_ID,
+      email: 'randomrandomtest@example.com',
+    } as IPopulatedUser
+
+    const MOCK_SCRUBBED_FORM = ({
+      _id: MOCK_FORM_ID,
+      title: 'mock title',
+      admin: { _id: MOCK_USER_ID },
+    } as unknown) as PublicForm
+
+    const BASE_FORM = {
+      admin: MOCK_USER,
+      _id: MOCK_FORM_ID,
+      title: MOCK_SCRUBBED_FORM.title,
+      getUniqueMyInfoAttrs: jest.fn().mockReturnValue([MyInfoAttribute.Name]),
+      getPublicView: jest.fn().mockReturnValue(MOCK_SCRUBBED_FORM),
```

Comment:
rather than mocking it manually, it might be better to use `dbHandler.insertEmailForm` and get an actual Form document

Reply:
maybe not leh, cause then we would need to init the db on this entire test suite. we have always (? fact check me) been mocking models for tests on the controller layer

Reply:
imo the mocking part is not super important because this test suite doesn't check that any of these methods are called anyway. so using an actual Form document just makes all the types cleaner, since you don't have to keep casting everything to `IFormSchema`, plus in future you don't have to keep adding mock methods to the mocked form. but am okay either way